### ISSUE TITLE
vagrant up fails when using smb as shared folder type

### ISSUE THREAD

**Issue by ghost** [puphpet/puphpet#2354]
Hey,

vagrant up fails for me when using smb as shared folder type. Host system is windows 7.
## relevant parts of the log

==> foo: Error: Failed to set owner to '0': Permission denied @ lchown_internal - /var/www
==> foo: Error: /Stage[main]/Puphpet_nginx/File[/var/www]/owner: change from www-data to root failed: Failed to set owner to '0': Permission denied @ lchown_internal - /var/www
==> foo: Error: Could not prefetch mysql_grant provider 'mysql': Command mysql is missing
==> foo: Error: Could not find a suitable provider for mysql_user
==> foo: Error: Could not find a suitable provider for rvm_system_ruby
==> foo: Error: Could not find a suitable provider for rvm_gem
## config.yml

```
vagrantfile:
    target: local
    vm:
        provider:
            local:
                box: puphpet/ubuntu1404-x64
                box_url: puphpet/ubuntu1404-x64
                box_version: '0'
                chosen_virtualizer: virtualbox
                virtualizers:
                    virtualbox:
                        modifyvm:
                            natdnshostresolver1: false
                        showgui: 0
                    vmware:
                        numvcpus: 1
                    parallels:
                        use_linked_clone: 0
                        check_guest_tools: 0
                        update_guest_tools: 0
                machines:
                    vflm_hna0ig3ejssr:
                        id: foo
                        hostname: foo
                        network:
                            private_network: 192.168.56.110
                            forwarded_port:
                                vflmnfp_vid4hb3u4oql:
                                    host: '7484'
                                    guest: '22'
                        memory: '2048'
                        cpus: '2'
        provision:
            puppet:
                manifests_path: puphpet/puppet/manifests
                module_path: puphpet/puppet/modules
                options:
                    - '--verbose'
                    - '--hiera_config /vagrant/puphpet/puppet/hiera.yaml'
        synced_folder:
            vflsf_af2o5cdrveul:
                source: ../
                target: /var/www
                sync_type: smb
                smb:
                    smb_host: ~
                    smb_username: ~
                    smb_password: ~
                    mount_options:
                        dir_mode: '0775'
                        file_mode: '0664'
                rsync:
                    args:
                        - '--verbose'
                        - '--archive'
                        - '-z'
                    exclude:
                        - .vagrant/
                        - .git/
                    auto: 'true'
                owner: www-data
                group: www-data
        usable_port_range:
            start: 10200
            stop: 10500
        post_up_message: ''
    ssh:
        host: 'false'
        port: 'false'
        private_key_path: 'false'
        username: vagrant
        guest_port: 'false'
        keep_alive: '1'
        forward_agent: 'false'
        forward_x11: 'false'
        shell: 'bash -l'
        insert_key: 'false'
    vagrant:
        host: detect
    proxy:
        http: ''
        https: ''
        ftp: ''
        no_proxy: ''
server:
    install: '1'
    packages:
        - mc
        - vim
        - htop
users_groups:
    install: '1'
    groups: {  }
    users: {  }
locale:
    install: '1'
    settings:
        default_locale: ''
        locales:
            - de_DE.UTF-8
            - en_GB.UTF-8
            - en_US.UTF-8
        timezone: Europe/Berlin
firewall:
    install: '1'
    rules: {  }
cron:
    install: '1'
    jobs: {  }
nginx:
    install: '1'
    settings:
        default_vhost: 1
        proxy_buffer_size: 128k
        proxy_buffers: '4 256k'
        proxy_connect_timeout: 600s
        proxy_send_timeout: 600s
        proxy_read_timeout: 600s
        names_hash_bucket_size: 128
    upstreams: {  }
    vhosts:
        nxv_maqydzxq1djz:
            server_name: foo.bar
            server_aliases:
                - www.foo.bar
            www_root: /var/www/foo/web
            listen_port: '80'
            index_files:
                - index.html
                - index.htm
                - index.php
            client_max_body_size: 1m
            ssl: '0'
            ssl_cert: ''
            ssl_key: ''
            ssl_port: '443'
            ssl_protocols: ''
            ssl_ciphers: ''
            rewrite_to_https: '1'
            spdy: '1'
            locations:
                nxvl_j3a57ostfqsc:
                    location: /
                    try_files:
                        - $uri
                        - /app.php$is_args$args
                    fastcgi: ''
                    fastcgi_index: ''
                    fastcgi_split_path: ''
                    proxy: ''
                    proxy_redirect: ''
                nxvl_q2yflmeldbtf:
                    location: '~ ^/(app_dev|config)\.php(/|$)'
                    fastcgi: '127.0.0.1:9000'
                    fastcgi_index: app_dev.php
                    fastcgi_split_path: '^(.+\.php)(/.*)$'
                    proxy: ''
                    proxy_redirect: ''
                nxvl_o85knsjumduy:
                    location: '~ ^/app\.php(/|$)'
                    fastcgi: '127.0.0.1:9000'
                    fastcgi_index: app.php
                    fastcgi_split_path: '^(.+\.php)(/.*)$'
                    proxy: ''
                    proxy_redirect: ''
    proxies: {  }
apache:
    install: '0'
    settings:
        version: 2.4
        user: www-data
        group: www-data
        default_vhost: true
        manage_user: false
        manage_group: false
        sendfile: 0
    modules:
        - proxy_fcgi
        - rewrite
    vhosts:
        av_ec3kxe8d4yfp:
            servername: foo.bar
            docroot: /var/www/foo/web
            port: '80'
            setenv:
                - 'APP_ENV dev'
            setenvif:
                - 'Authorization "(.*)" HTTP_AUTHORIZATION=$1'
            custom_fragment: ''
            ssl: '0'
            ssl_cert: ''
            ssl_key: ''
            ssl_chain: ''
            ssl_certs_dir: ''
            ssl_protocol: ''
            ssl_cipher: ''
            directories:
                avd_4tqzlg55tbe7:
                    path: /var/www/foo/web
                    options:
                        - Indexes
                        - FollowSymlinks
                        - MultiViews
                    allow_override:
                        - All
                    require:
                        - 'all granted'
                    custom_fragment: ''
                    files_match:
                        avdfm_mjv8l66w2y0v:
                            path: \.php$
                            sethandler: 'proxy:fcgi://127.0.0.1:9000'
                            custom_fragment: ''
                            provider: filesmatch
                    provider: directory
letsencrypt:
    install: '1'
    settings:
        email: ''
    domains: {  }
php:
    install: '1'
    settings:
        version: '70'
    modules:
        php:
            - cli
            - intl
            - curl
            - memcached
        pear: {  }
        pecl: {  }
    ini:
        display_errors: 'On'
        error_reporting: '-1'
        session.save_path: /var/lib/php/session
        date.timezone: Europe/Berlin
    fpm_ini:
        error_log: /var/log/php-fpm.log
    fpm_pools:
        phpfp_k29h0kudl3iu:
            ini:
                prefix: www
                listen: '127.0.0.1:9000'
                security.limit_extensions: .php
                group: www-data
                user: www-data
    composer: '1'
    composer_home: ''
xdebug:
    install: '0'
    settings:
        xdebug.default_enable: '1'
        xdebug.remote_autostart: '0'
        xdebug.remote_connect_back: '1'
        xdebug.remote_enable: '1'
        xdebug.remote_handler: dbgp
        xdebug.remote_port: '9000'
blackfire:
    install: '0'
    settings:
        server_id: 01ea3680-ab0b-4843-ba85-2e931d417f82
        server_token: 0a5b3cdafdb7d4b58f1c571192f615f46a5bf522727d043a9970c5629b225635
        agent:
            http_proxy: ''
            https_proxy: ''
            log_file: stderr
            log_level: '1'
        php:
            agent_timeout: '0.25'
            log_file: ''
            log_level: '1'
xhprof:
    install: '0'
wpcli:
    install: '0'
    version: v0.21.1
drush:
    install: '0'
    version: 8.0.5
ruby:
    install: '1'
    versions:
        rv_08v2ti8g4t5p:
            default: '1'
            bundler: '1'
            version: 1.9.3
            gems:
                - deep_merge
                - activesupport
                - vine
python:
    install: '1'
    packages: {  }
    versions: {  }
nodejs:
    install: '0'
    settings:
        version: '5'
    npm_packages: {  }
hhvm:
    install: '0'
    nightly: 0
    composer: '1'
    composer_home: ''
    settings: {  }
    server_ini:
        hhvm.server.host: 127.0.0.1
        hhvm.server.port: '9000'
        hhvm.log.use_log_file: '1'
        hhvm.log.file: /var/log/hhvm/error.log
    php_ini:
        display_errors: 'On'
        error_reporting: '-1'
        date.timezone: UTC
mariadb:
    install: '1'
    settings:
        version: '10.1'
        root_password: '1'
        override_options: {  }
    adminer: 0
    users:
        mariadbnu_vyvwvd9m3gjb:
            name: foo
            password: '1'
    databases: {  }
    grants:
        mariadbng_gf8r47fpy6la:
            user: foo
            table: '*.*'
            privileges:
                - ALL
mysql:
    install: '0'
    settings:
        version: '5.7'
        root_password: '1'
        override_options: {  }
    adminer: 0
    users:
        mysqlnu_6soou3trwx8s:
            name: foo
            password: '1'
    databases: {  }
    grants:
        mysqlng_k43900gndxjm:
            user: foo
            table: '*.*'
            privileges:
                - ALL
postgresql:
    install: '0'
    settings:
        global:
            encoding: UTF8
            version: '9.4'
        server:
            postgres_password: '1'
    databases: {  }
    users: {  }
    grants: {  }
    adminer: 0
mongodb:
    install: '0'
    settings:
        bind_ip: 127.0.0.1
        port: '27017'
    globals:
        version: 2.6.0
    databases: {  }
redis:
    install: '1'
    settings:
        port: '6379'
sqlite:
    install: '0'
    adminer: 0
    databases: {  }
mailhog:
    install: '1'
    settings:
        smtp_ip: 0.0.0.0
        smtp_port: 1025
        http_ip: 0.0.0.0
        http_port: '8025'
        path: /usr/local/bin/mailhog
beanstalkd:
    install: '0'
    settings:
        listenaddress: 0.0.0.0
        listenport: '11300'
        maxjobsize: '65535'
        maxconnections: '1024'
        binlogdir: /var/lib/beanstalkd/binlog
        binlogfsync: null
        binlogsize: '10485760'
    beanstalk_console: 0
rabbitmq:
    install: '0'
    settings:
        port: '5672'
    users: {  }
    vhosts: {  }
    plugins: {  }
elastic_search:
    install: '1'
    settings:
        version: 2.3.2
        java_install: true
        manage_repo: true
        repo_version: '2.x'
    instances:
        esi_q38c9cgd4d08:
            name: foo
solr:
    install: '0'
    settings:
        version: 4.10.2
        port: '8984'
```

Has anyone else the same problem?

Best
Christian

**Comment by csiewert** [comment]
According to https://github.com/alehkot/druphpet/issues/30 I tried to set the 

    folder['smb']['mount_options']['file_mode']

to 0764 but the same errors appears.

**Comment by csiewert** [comment]
For now, I switched to NFS with this plugin: https://github.com/winnfsd/vagrant-winnfsd Performance of NFS is awesome!

**Comment by jtreminio** [comment]
Will be fixed by #1837
